torch/_dynamo/variables/higher_order_ops.py

```diff
@@ -2514,6 +2516,76 @@ def call_function(
         return _make_inlined(tx, pytree.tree_unflatten)(variable, treespec)
 
 
+# Copy paste of the above class, but enables an arbitrary callable
+class WrapGenericHigherOrderVariable(WrapHigherOrderVariable):
+    def __init__(self, hop, source) -> None:
+        super().__init__(hop, source)
+
+    def call_function(
+        self,
+        tx: "InstructionTranslator",
+        args: list[VariableTracker],
+        kwargs: dict[str, VariableTracker],
+    ) -> VariableTracker:
+        from .builder import wrap_fx_proxy
+
+        func_var = kwargs["wrapper_fn"]
+
+        if isinstance(func_var, torch._dynamo.variables.UserFunctionVariable):
+            func = func_var.fn
+        elif isinstance(
+            func_var, torch._dynamo.variables.functions.FunctoolsPartialVariable
+        ):
+            func = func_var.as_python_constant()
+        else:
+            raise RuntimeError(
+                f"WrapGenericHigherOrderVariable: Unsupported function {type(func_var)}"
+            )
+        gmod_kwargs = {
+            name: kwargs[name] for name in kwargs.keys() if name != "wrapper_fn"
+        }
+        (
+            p_args,
+            _,
+            example_value,
+            _body_r,
+            treespec,
+            gmod,
+            _,
+        ) = self.create_wrapped_node(
```

Comment:
Can we change the name of the hop from WrapGeneric to a more informative one? It seems that the purpose of the hop is to bypass tracing of a wrapper function while tracing the wrapped function. So maybe something like BypassingWrapper or something similar?